File: main/number-theory/gcd.cpp
/*
gcd.cpp
sidb95
bhatoresiddharth@gmail.com 
11 July 2024
*/

#include <iostream>

class Solution {
public:
    FLAG = false;
    
    Solution() {
        FLAG = true;
    }

    ~Solution() {
        FLAG = false;
    }

    int gcd(int a, int b) {
        if (a == 1) {
            return b;
        }
        else {
            int TEMP = a;
            a = b;
            b = a % b;
        }
        return gcd(a, b);
    }
};

File: main/core-python/number-theory/gcd.py
"""
gcd.py
sidb95
bhatoresiddharth@gmail.com 
12 July 2024
"""

class Solution:
  def __init__(self):
    pass

  def gcd(a: int, b: int) -> int:
    if (a == 1):
      return b
    else:
      TEMP = a
      a = a % b
      b = a
      return gcd(a, b)
